cloudinstall/machinewait.py

```diff
@@ -124,6 +124,7 @@ def update(self):
         # ensure that the button is always focused:
         self.main_pile.focus_position = len(self.main_pile.contents) - 1
 
+    # XXX: Is this utils.async needed?
```

Comment:
That's what this PR is supposed to address, if headless mode we don't utilize the ThreadPool otherwise we just submit the function to the pool